app/components/primer/anchored_position.ts

```diff
@@ -0,0 +1,158 @@
+import type {AnchorAlignment, AnchorSide, PositionSettings} from '@primer/behaviors'
+import {getAnchoredPosition} from '@primer/behaviors'
+
+const updateWhenVisible = (() => {
+  const anchors = new Set<AnchoredPositionElement>()
+  let intersectionObserver: IntersectionObserver | null = null
+  function updateAnchorsOnResize() {
+    for (const anchor of anchors) {
+      anchor.update()
+    }
+  }
+  return (el: AnchoredPositionElement) => {
+    // eslint-disable-next-line github/prefer-observers
+    el.ownerDocument.addEventListener('resize', updateAnchorsOnResize)
+    window.addEventListener('resize', updateAnchorsOnResize)
+    intersectionObserver ||= new IntersectionObserver(entries => {
+      for (const entry of entries) {
+        const target = entry.target as AnchoredPositionElement
+        if (entry.isIntersecting) {
+          target.update()
+          anchors.add(target)
+        } else {
+          anchors.delete(target)
+        }
+      }
+    })
+    intersectionObserver.observe(el)
+  }
+})()
+
+export default class AnchoredPositionElement extends HTMLElement implements PositionSettings {
```

Comment:
This is called `AnchoredPosition` as it only has the anchoring behaviour. We can re-use this for tooltip, for example.